src/Compilers/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -23,6 +23,9 @@ Microsoft.CodeAnalysis.Operations.OperationWalker<TArgument>.OperationWalker() -
 Microsoft.CodeAnalysis.SymbolDisplayPartKind.RecordClassName = 31 -> Microsoft.CodeAnalysis.SymbolDisplayPartKind
 const Microsoft.CodeAnalysis.WellKnownDiagnosticTags.CompilationEnd = "CompilationEnd" -> string!
 Microsoft.CodeAnalysis.SyntaxContextReceiverCreator
+Microsoft.CodeAnalysis.SyntaxNode.IsIncrementallyIdenticalTo(Microsoft.CodeAnalysis.SyntaxNode! other) -> bool
+Microsoft.CodeAnalysis.SyntaxNodeOrToken.IsIncrementallyIdenticalTo(Microsoft.CodeAnalysis.SyntaxNodeOrToken other) -> bool
+Microsoft.CodeAnalysis.SyntaxToken.IsIncrementallyIdenticalTo(Microsoft.CodeAnalysis.SyntaxToken token) -> bool
```

Comment:
@dotnet/roslyn-compiler  these are new APIs i'm adding to effectively ask "do these nodes have identical green nodes".  It's a way to dtermine *extremely* quickly what parts of a tree were *definitely* reused across an incremental edit..

This is used in this PR to take two trees, and conservatively (not minimally) determine the portion of the tree that changed.  For any green nodes that are identical between the before/after we know that that didn't change.  So we only need to examine nodes that did change.  See the logic in SyntacticChangeRangeComputer below for how we utilize this.

Reply:
If these were added to the public API, I would prefer them be added as static methods in a helper class which are _not_ extension methods so they aren't showing up for users working with these types in common scenarios.

Reply:
It's a fair point. But I'm not sure a good static class these should go to. There is also precedence here as these three types already have IsEquivalentTo. So this really is a sibling method to those. But I would be fine moving these of people have a good suggestion on where it should go

Reply:
Honestly, this seems fine to me as a location for the API. As Cyrus mentioned, we already have `IsEquivalentTo` APIs, and those aren't common-case APIs either.